core/corehttp/commands.go

```diff
@@ -140,15 +141,15 @@ func commandsOption(cctx oldcmds.Context, command *cmds.Command) ServeOption {
 }
 
 // CommandsOption constructs a ServerOption for hooking the commands into the
-// HTTP server.
+// HTTP server. These can be called only with POST, "OPTIONS" and "HEAD".
 func CommandsOption(cctx oldcmds.Context) ServeOption {
-	return commandsOption(cctx, corecommands.Root)
+	return commandsOption(cctx, corecommands.Root, []string{"POST", "OPTIONS", "HEAD"})
```

Comment:
What is the rationale for allowing `OPTIONS` and `HEAD` on write opts?

Also do we not recognize `PUT`/`DELETE` anywhere?

Reply:
Ok, I was tired. I have removed HEAD (as this is akin to GET).

OPTIONS is necessary for CORS reasons, as it is how browsers figure out what's allowed and what not.

PUT/DELETE are not handled (from now on). In this API all request methods do the same thing.